File: attachment-microservice/src/main/scala/data/entity/Attachment.scala
package data.entity

import javax.persistence.{Column, Entity, Table}

import scala.beans.BeanProperty

/**
  * 附件实体类
  *
  * @author 刘佳兴
  * @version V1.0
  */
@Entity
@Table(name = "attachment")
class Attachment extends BaseEntity {

  //UUID名称
  @BeanProperty
  var name: String = _

  //真实名称
  @BeanProperty
  var realName: String = _

  //上传的相对路径
  @BeanProperty
  var relativePath: String = _

  //后缀名
  @Column(length = 30)
  @BeanProperty
  var contentType: String = _

  //文件大小
  @BeanProperty
  var size: Long = _

  //后缀名
  @BeanProperty
  var suffix: String = _

  def this(id: Long) {
    this()
    this.id = id
  }

  def this(name: String, realName: String, relativePath: String, contentType: String, size: Long, suffix: String) {
    this()
    this.name = name
    this.realName = realName
    this.relativePath = relativePath
    this.contentType = contentType
    this.suffix = suffix
    this.size = size
  }

  override def toString = s"Attachment($name, $realName, $relativePath, $contentType, $size, $suffix)"
}

File: attachment-microservice/src/main/java/data/entity/Attachment.java
package data.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Transient;
import java.util.ArrayList;
import java.util.List;

/**
 * @author 刘佳兴
 */
@Data
@Entity
@Table(name = "attachment")
@EqualsAndHashCode(callSuper = true)
public class Attachment extends BaseEntity {

    //UUID名称
    protected String name;

    //真实名称
    protected String realName;

    //上传的相对路径
    protected String relativePath;

    //文件类型
    @Column(length = 30)
    protected String contentType;

    //文件大小
    protected  Long size;

    @Transient
    protected String url;

    public Attachment() {

    }

    public Attachment(String name, String realName, String relativePath, String contentType, Long size) {
        this.name = name;
        this.realName = realName;
        this.relativePath = relativePath;
        this.contentType = contentType;
        this.size = size;
    }
}
